Dropdown: allow onChanged to be called when an item has been reselected

janormanms opened the pull request:
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #6759 
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Dropdown only notifies when an item is clicked if the selection has actually changed. Adding a parameter to allow onChanged to be called when an item is reselected in single select mode

#### Focus areas to test

Dropdown

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/6789)

cliffkoh commented on packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.test.tsx:
```diff
@@ -161,6 +161,30 @@ describe('Dropdown', () => {
+
+      document.body.appendChild(container);
+
+      const onChangeSpy = jasmine.createSpy('onChange');
```
Just for consistency, could you use jest.fn() for the spy instead? (I'm honestly surprised that jasmine is in our dependency tree :))

janormanms replied:
Sounds good, updated all of the instances in this file

cliffkoh commented on packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.test.tsx:
```diff
@@ -161,6 +161,30 @@ describe('Dropdown', () => {
       }
     });
 
+    it('issues the onChange callback when the selected item is the same if notifyOnReselect is true', () => {
```
:+1:

cliffkoh commented:
Forgot to say, we should also have a test for the current behavior (i,e. it does not reselect if the opt in is false)

janormanms commented:
Thanks! I'll update to use jest.fn() shortly. And there is already a test for the current behavior a few down from the one I added, initially I had added one as well but then realized it was a duplicate after reading the other tests

cliffkoh commented:
👏 Looks good to be checked in. I'll keep this open a while longer to allow more people to review and merge this before end of day, in time for next publish.

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/OfficeDev/office-ui-fabric-react?pullRequest=6789) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/OfficeDev/office-ui-fabric-react?pullRequest=6789) before we can accept your contribution.<br/><br/>:x: janormanms [sign now](https://cla.opensource.microsoft.com/OfficeDev/office-ui-fabric-react?pullRequest=6789)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/OfficeDev/office-ui-fabric-react?pullRequest=6789) it.</sub>

janormanms commented:
@cliffkoh , I didn't see anything specific to my changes in the check failures, though with this being my first change on here maybe I missed where to look. Is there a way to rerun the deploy and other checks? One noted an Azure upload error though I don't know if that's a symptom or a root cause.

Similar issues with my other PR: #6797

cliffkoh commented:
Let me close and reopen to trigger a rebuild. PR deploy should be working by now.
